.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,34 @@
+name: pull request checks
+
+on:
+  pull_request:
+    branches: ['*']
+
+jobs:
+  build-test:
+    name: Build, lint, test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ['1.21']
+
+    steps:
+    - name: install make
+      run: sudo apt-get install make
+    - name: set up go 1.x
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ matrix.go }}
+    - name: checkout
+      uses: actions/checkout@v3
+    - name: run build
+      run: make build && git diff --exit-code
+    - name: run unit tests
+      run: make test
+    - name: Report coverage
+      uses: codecov/codecov-action@v2.1.0
+      with:
+        files: ./cover.out
+        flags: unittests
+        fail_ci_if_error: false 
+        args: --timeout 5m0s
```

Comment:
@msherif1234 I'm getting a timeout here while linting (part of make build).
Any idea what is going on ?
https://github.com/netobserv/network-observability-cli/actions/runs/8236289627/job/22522495525

It's super fast locally